arangod/RocksDBEngine/RocksDBEngine.h

```diff
@@ -646,6 +647,16 @@ class RocksDBEngine final : public StorageEngine {
 
   // @brief persistor for replicated logs
   std::shared_ptr<RocksDBLogPersistor> _logPersistor;
+
+  // Checksum env for when creation of sha files is enabled
+  // this is for when encryption is enabled, sha files will be created
+  // after the encryption of the .sst files
+  std::unique_ptr<rocksdb::Env> _checksumEnv;
+
+  rocksdb::Env* NewChecksumEnv(rocksdb::Env* base_env,
```

Comment:
I also suggest that we make this function return a unique_ptr, and not a raw ptr.